numpy/core/src/multiarray/compiled_base.c

```diff
@@ -1530,114 +1495,11 @@ pack_inner(const char *inptr,
-            v = _mm_cmpeq_epi8(v, zero);
-            v = _mm_cmpeq_epi8(v, zero);
-            /* extract msb of 16 bytes and pack it into 16 bit */
-            r = _mm_movemask_epi8(v);
-            /* store result */
-            memcpy(outptr, &r, 1);
-            outptr += out_stride;
-            memcpy(outptr, (char*)&r + 1, 1);
-            outptr += out_stride;
-            inptr += 16;
-        }
-    }
-#elif defined NPY_HAVE_NEON
-    if (in_stride == 1 && element_size == 1 && n_out > 2) {
-        /* don't handle non-full 8-byte remainder */
-        npy_intp vn_out = n_out - (remain ? 1 : 0);
-        vn_out -= (vn_out & 1);
-        for (index = 0; index < vn_out; index += 2) {
-            unsigned int r;
-            npy_uint64 a = *((uint64_unaligned*)inptr);
-            npy_uint64 b = *((uint64_unaligned*)(inptr + 8));
-            if (order == 'b') {
-                a = npy_bswap8(a);
-                b = npy_bswap8(b);
-            }
-            uint64x2_t v = vcombine_u64(vcreate_u64(a), vcreate_u64(b));
-            uint64x2_t zero = vdupq_n_u64(0);
-            /* false -> 0x00 and true -> 0xFF */
-            v = vreinterpretq_u64_u8(vmvnq_u8(vceqq_u8(vreinterpretq_u8_u64(v), vreinterpretq_u8_u64(zero))));
-            /* extract msb of 16 bytes and pack it into 16 bit */
-            uint8x16_t input = vreinterpretq_u8_u64(v);
-            r = sign_mask(input);
-            /* store result */
-            memcpy(outptr, &r, 1);
-            outptr += out_stride;
-            memcpy(outptr, (char*)&r + 1, 1);
-            outptr += out_stride;
-            inptr += 16;
-        }
-    }
-#endif
-
-    if (remain == 0) {                  /* assumes n_in > 0 */
-        remain = 8;
-    }
-    /* Don't reset index. Just handle remainder of above block */
-    for (; index < n_out; index++) {
-        unsigned char build = 0;
-        int i, maxi;
-        npy_intp j;
-
-        maxi = (index == n_out - 1) ? remain : 8;
-        if (order == 'b') {
-            for (i = 0; i < maxi; i++) {
-                build <<= 1;
-                for (j = 0; j < element_size; j++) {
-                    build |= (inptr[j] != 0);
-                }
-                inptr += in_stride;
-            }
-            if (index == n_out - 1) {
-                build <<= 8 - remain;
-            }
-        }
-        else
-        {
-            for (i = 0; i < maxi; i++) {
-                build >>= 1;
-                for (j = 0; j < element_size; j++) {
-                    build |= (inptr[j] != 0) ? 128 : 0;
-                }
-                inptr += in_stride;
-            }
-            if (index == n_out - 1) {
-                build >>= 8 - remain;
-            }
-        }
-        *outptr = (char)build;
-        outptr += out_stride;
-    }
+    #ifndef NPY_DISABLE_OPTIMIZATION
+        #include "compiled_base.dispatch.h"
+    #endif
+    NPY_CPU_DISPATCH_CALL(return compiled_base_pack_inner,
```

Comment:
```suggestion
    NPY_CPU_DISPATCH_CALL(compiled_base_pack_inner,
```
Return void!

Reply:
Would it better if you used `simd_` as a prefix instead of `compiled_`?

Reply:
I refer to the previous `einsum `named case, with `module_function` format, I think we can just add `simd_` as a prefix, the new format turns to `simd_module_function`